addons/website/models/res_config.py

```diff
@@ -54,6 +55,24 @@ def _default_website(self):
     has_google_analytics = fields.Boolean("Google Analytics")
     has_google_analytics_dashboard = fields.Boolean("Google Analytics in Dashboard")
     has_google_maps = fields.Boolean("Google Maps")
+    auth_signup_uninvited = fields.Selection([
+        ('b2b', 'On invitation (B2B)'),
+        ('b2c', 'Free sign up (B2C)'),
+    ], string='Customer Account')
+
+    @api.model
+    def get_default_auth_signup_uninvited(self, fields):
+        get_param = self.env['ir.config_parameter'].get_param
+        # we use safe_eval on the result, since the value of the parameter is a nonempty string
+        allow_uninvited = safe_eval(get_param('auth_signup.allow_uninvited', 'False'))
+        return {
+            'auth_signup_uninvited': 'b2c' if allow_uninvited else 'b2b',
+        }
+
+    def set_auth_signup_uninvited(self):
+        set_param = self.env['ir.config_parameter'].set_param
```

Comment:
You forgot to ask the same question for get_param just above.

It's to avoid having a long line next (>100 characters). Okay, we uselessly define an extra variable, but it's more readable this way imo.